### PR TITLE
ContactUsForm Component #35

### LINKED ISSUE #35: Contact page
Build contact page according to figma design, create contact form, check netfliy form submission and connect it to firebase(optional), build tests, make it responsive.

### PR DESCRIPTION
**Pull request by ssinikoglu**
This PR consists of ContactUsForm component of Contact page. I created new PR since I changed all the format and documentation of the contact form (NetlifyForm file has been deleted now.). The PR is written according to via [this](https://www.netlify.com/blog/2017/07/20/how-to-integrate-netlifys-form-handling-in-a-react-app/) Netlify documentation. 

Upcoming PRs for the Contact Page will be ContactUsBottom (the component has only visual features (curve and images) and responsive design.), Contact Container.

Please review. I am looking forward to your comments.

### CODE REVIEW

File: src/components/ContactUsForm/index.jsx
Comment (louisrli) [inline src/components/ContactUsForm/index.jsx]: since this is a "pure function" (can google it) that is independent from the component i'd say it actually makes more sense to move this below the imports (in the same file), but outside the component. and call it "encodeFormData"

Comment (louisrli) [reply src/components/ContactUsForm/index.jsx]: separating functions like that makes it clear to the reader that it has no dependency on anything inside the component.

File: src/components/ContactUsForm/index.jsx
Comment (louisrli) [inline src/components/ContactUsForm/index.jsx]: this timeout is weird. you probably don't want to do this, and 5 seconds is a long time for the user to wait. you may want a state called isLoading or something (ibrahim did something similar) but in general we never write code with manual timeouts like this. it could take, in reality, longer than 5 seconds.

what's the reason?

File: src/components/ContactUsForm/index.jsx
Comment (louisrli) [inline src/components/ContactUsForm/index.jsx]: generally we never console.log, prefer to alert(error) here if you want a low fidelity error, otherwise
```
const [submitError, setSubmitError] = React.useState(null)

.catch((error) => setSubmitError(error))

<div class="errorMessageDiv">{submitError}</div>
```

something like that

File: src/components/ContactUsForm/index.jsx
Comment (louisrli) [inline src/components/ContactUsForm/index.jsx]: style={{visibility: isSubmitted ? "visible" : ... }}

File: src/components/ContactUsForm/index.jsx
Comment (louisrli) [inline src/components/ContactUsForm/index.jsx]: i would call this contactUsForm.postSubmissionMessage to be super clear

Comment (louisrli) [reply src/components/ContactUsForm/index.jsx]: post meaning "after"

File: src/components/ContactUsForm/index.jsx
Comment (louisrli) [inline src/components/ContactUsForm/index.jsx]: this line should be deletable

**Review by louisrli — COMMENTED** [review]
mostly looks good actually. nice work following the netlify docs. some minor comments

File: src/components/ContactUsForm/index.jsx
Comment (ssinikoglu) [reply src/components/ContactUsForm/index.jsx]: I removed the line, PTAL.

File: src/components/ContactUsForm/index.jsx
Comment (ssinikoglu) [reply src/components/ContactUsForm/index.jsx]: All submissionMessage related names are changed, PTAL.

File: src/components/ContactUsForm/index.jsx
Comment (ssinikoglu) [reply src/components/ContactUsForm/index.jsx]: Thanks for your review, it's format changed and become cleaner, PTAL.

File: src/components/ContactUsForm/index.jsx
Comment (ssinikoglu) [reply src/components/ContactUsForm/index.jsx]: I changed console.log into alert for now.

File: src/components/ContactUsForm/index.jsx
Comment (ssinikoglu) [reply src/components/ContactUsForm/index.jsx]: The main reason I have used timeout (via using isSubmitted state) is that showing post submit message a little more to the user. I gave 5 seconds by trying and it seems I underestimated back sides of it. I'm looking Ibrahim's code, I'll try adapting it to  handleSubmit function.

File: src/components/ContactUsForm/index.jsx
Comment (ssinikoglu) [reply src/components/ContactUsForm/index.jsx]: Thanks for reviewing and explanation. I moved the function to the new file. PTAL

File: package.json
Comment (louisrli) [inline package.json]: is this intended to be deleted?

File: src/components/ContactUsForm/encodeFormData.jsx
Comment (louisrli) [inline src/components/ContactUsForm/encodeFormData.jsx]: actually it's cleaner to put this in the same file as.the component. under the imports. it's harder for readability when one function is in its own file because people need to jump files to read it

**Review by louisrli — CHANGES_REQUESTED** [review]
looks good. will approve when tests pass

File: package.json
Comment (ssinikoglu) [reply package.json]: It is not comletely deleted, version of the icons changed after latest merge.

File: src/components/ContactUsForm/encodeFormData.jsx
Comment (ssinikoglu) [reply src/components/ContactUsForm/encodeFormData.jsx]: I moved the function under the component file. PTAL

**Comment by ssinikoglu** [comment]
> looks good. will approve when tests pass

@majdajroudi fixed the problem, all tests are passing now.

**Review by louisrli — APPROVED** [review]

File: src/components/ContactUsForm/index.jsx
Comment (tmansuroglu) [inline src/components/ContactUsForm/index.jsx:51]: if you change className to "contactUsForm" all class names will be correct

**Review by tmansuroglu — APPROVED** [review]
Nice work!

**Comment by ssinikoglu** [comment]
Hi @Ammar-64, PR is approved, PTAL.

**Pull request merged**
